src/Simplex/Messaging/Agent/Client.hs

```diff
@@ -264,23 +269,43 @@ deleteQueue c RcvQueue {server, rcvId, rcvPrivateKey} =
     deleteSMPQueue smp rcvPrivateKey rcvId
 
 sendAgentMessage :: AgentMonad m => AgentClient -> SndQueue -> SenderTimestamp -> AMessage -> m ()
-sendAgentMessage c SndQueue {server, sndId, sndPrivateKey, encryptKey} senderTs agentMsg = do
-  msg <- mkAgentMessage encryptKey senderTs agentMsg
-  withLogSMP c server sndId "SEND <message>" $ \smp ->
-    sendSMPMessage smp (Just sndPrivateKey) sndId msg
-
-mkAgentMessage :: AgentMonad m => EncryptionKey -> SenderTimestamp -> AMessage -> m ByteString
-mkAgentMessage encKey senderTs agentMessage = do
-  let msg =
-        serializeSMPMessage
-          SMPMessage
-            { senderMsgId = 0,
-              senderTimestamp = senderTs,
-              previousMsgHash = "1234", -- TODO hash of the previous message
-              agentMessage
-            }
-  paddedSize <- asks paddedMsgSize
-  liftError cryptoError $ C.encrypt encKey paddedSize msg
+sendAgentMessage c sq@SndQueue {server, sndId, sndPrivateKey} senderTs agentMsg =
+  withLogSMP_ c server sndId "SEND <message>" $ \smp -> do
+    msg <- mkAgentMessage smp sq senderTs agentMsg
+    liftSMP $ sendSMPMessage smp (Just sndPrivateKey) sndId msg
+
+mkAgentMessage :: AgentMonad m => SMPClient -> SndQueue -> SenderTimestamp -> AMessage -> m ByteString
+mkAgentMessage smp sq senderTs agentMessage = do
+  encryptMessage smp sq $
+    SMPMessage
+      { senderMsgId = 0,
+        senderTimestamp = senderTs,
+        previousMsgHash = "1234", -- TODO hash of the previous message
+        agentMessage
+      }
+
+encryptMessage :: AgentMonad m => SMPClient -> SndQueue -> SMPMessage -> m ByteString
+encryptMessage smp SndQueue {encryptKey, signKey} msg = do
+  paddedSize <- asks $ (blockSize smp -) . reservedMsgSize
+  liftError cryptoError $ do
+    enc <- C.encrypt encryptKey paddedSize $ serializeSMPMessage msg
+    C.Signature sig <- C.sign signKey enc
+    pure $ sig <> enc
```

Comment:
it's not necessary, but maybe move signature to `mkAgentMessage`? A bit strange to have signature in a method called `encrypt`, and it would mirror separation of `decrypt` and `verify`